## Update: Removed --dist=loadscope

**Issue Found:**
The `--dist=loadscope` change exposed a pre-existing test isolation issue in `test_token_counter.py`. The tokenizer cache persists between tests when they run on the same worker, causing:
```
AssertionError: Expected 'from_pretrained' to be called once. Called 0 times.
```

**Root Cause:**
When tests from the same file run together (loadscope behavior), the tokenizer gets cached from earlier tests, so later tests never call `from_pretrained`.

**Solution:**
Removed `--dist=loadscope` from this PR. The test isolation issue should be fixed separately by either:
1. Adding proper cache cleanup between tests
2. Using `@pytest.fixture(autouse=True)` to clear tokenizer caches

**What Remains:**
Still keeping the two valuable improvements that don't cause issues:
✅ Vertex AI tests isolated (workers: 1)  
✅ LLM tests reduced parallelism (workers: 2)

These alone should significantly improve test stability without the side effects.

--- a/.github/workflows/test-litellm-matrix.yml
+++ b/.github/workflows/test-litellm-matrix.yml
@@ -18,10 +18,15 @@ jobs:
       matrix:
         test-group:
           # tests/test_litellm split by subdirectory (~560 files total)
-          - name: "llms"
-            path: "tests/test_litellm/llms"
-            workers: 2  # Reduced from 4 to 2 to avoid race conditions
-            reruns: 2   # Retry flaky tests twice
+          # Vertex AI tests separated for better isolation (prevent auth/env pollution)
+          - name: "llms-vertex"
+            path: "tests/test_litellm/llms/vertex_ai"
+            workers: 1
+            reruns: 2
+          - name: "llms-other"
+            path: "tests/test_litellm/llms --ignore=tests/test_litellm/llms/vertex_ai"
+            workers: 2
+            reruns: 2
           # tests/test_litellm/proxy split by subdirectory (~180 files total)
           - name: "proxy-guardrails"
             path: "tests/test_litellm/proxy/guardrails tests/test_litellm/proxy/management_endpoints tests/test_litellm/proxy/management_helpers"
@@ -105,4 +110,5 @@ jobs:
             -n ${{ matrix.test-group.workers }} \
             --reruns ${{ matrix.test-group.reruns }} \
             --reruns-delay 1 \
+            --dist=loadscope \
             --durations=20

